appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-service/src/main/java/fish/payara/appserver/monitoring/rest/service/admin/GetRestMonitoringConfiguration.java

```diff
@@ -0,0 +1,127 @@
+    If you wish your version of this file to be governed by only the CDDL or
+    only the GPL Version 2, indicate your decision by adding "[Contributor]
+    elects to include this software in this distribution under the [CDDL or GPL
+    Version 2] license."  If you don't indicate a single choice of license, a
+    recipient has the option to distribute your version of this file under
+    either the CDDL, the GPL Version 2 or to extend the choice of license to
+    its licensees as provided above.  However, if you add GPL Version 2 code
+    and therefore, elected the GPL Version 2 license, then the option applies
+    only if the new code is made subject to such option by the copyright
+    holder.
+ */
+
+package fish.payara.appserver.monitoring.rest.service.admin;
+
+import com.sun.enterprise.config.serverbeans.Config;
+import com.sun.enterprise.util.ColumnFormatter;
+import com.sun.enterprise.util.SystemPropertyConstants;
+import fish.payara.appserver.monitoring.rest.service.configuration.RestMonitoringConfiguration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import javax.inject.Inject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.RestEndpoint;
+import org.glassfish.api.admin.RestEndpoints;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+
+/**
+ * ASAdmin command to get the rest monitoring services configuration
+ *
+ * @author michael ranaldo <michael.ranaldo@payara.fish>
+ */
+@Service(name = "get-rest-monitoring-configuration")
+@PerLookup
+@ExecuteOn({RuntimeType.DAS})
+@TargetType(value = {CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, 
+    CommandTarget.CLUSTERED_INSTANCE, CommandTarget.CONFIG})
+@RestEndpoints({
+    @RestEndpoint(configBean = RestMonitoringConfiguration.class,
+            opType = RestEndpoint.OpType.GET,
+            path = "get-rest-monitoring-configuration",
+            description = "Gets the Rest Monitoring Configuration")
+})
+public class GetRestMonitoringConfiguration implements AdminCommand {
+    
+    private final String OUTPUT_HEADERS[] = {"Enabled", "Rest Monitoring Application Name", "Context Root", "Security Enabled"};
+
+    @Inject
+    private Target targetUtil;
+
+    @Param(name = "target", defaultValue = SystemPropertyConstants.DAS_SERVER_NAME, optional = true)
+    private String target;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        Config config = targetUtil.getConfig(target);
+        
+        RestMonitoringConfiguration restMonitoringConfiguration = config
```

Comment:
Correct me if I am wrong, but it looks NPE is possible here ? I guess this row should be below the null check for the config variable?

Reply:
Good spot! I agree this should be below the null check.